etcdserver/cluster.go

```diff
@@ -334,13 +334,23 @@ func (c *cluster) UpdateAttributes(id types.ID, attr Attributes) bool {
 		m.Attributes = attr
 		return true
 	}
+	b, err := json.Marshal(attr)
+	if err != nil {
+		plog.Panicf("marshal attributes should never fail: %v", err)
+	}
+	p := path.Join(memberStoreKey(id), attributesSuffix)
 	_, ok := c.removed[id]
 	if ok {
 		plog.Warningf("skipped updating attributes of removed member %s", id)
```

Comment:
the if/else nesting can be helped a bit by `return false` after the `Warningf()`, then removing the following `else`